### ISSUE TITLE
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter

### ISSUE THREAD

**Issue by pichfl** [ember-template-lint/ember-cli-template-lint#345]
This happens when running ember-cli-template-lint in a project with ember-cli > 3.2.0. 

```
[API] Warning: The .read and .rebuild APIs will stop working in the next Broccoli version
[API] Warning: Use broccoli-plugin instead: https://github.com/broccolijs/broccoli-plugin
[API] Warning: Plugin uses .read/.rebuild API: Filter
[API] Warning: Plugin uses .read/.rebuild API: Filter
[API] Warning: Plugin uses .read/.rebuild API: Filter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
```

**Comment by rwjblue** [comment]
Thank you for reporting!

**Comment by jrjohnson** [comment]
And now the next Broccoli version. When I attempt to run `EMBER_CLI_BROCCOLI_2=true ember build` I get the following error:

```
TemplateLinter: The .read/.rebuild API is no longer supported as of Broccoli 1.0. Plugins must now derive from broccoli-plugin. https://github.com/broccolijs/broccoli/blob/master/docs/broccoli-1-0-plugin-api.md
```

**Comment by rwjblue** [comment]
This is fixed on master and will soon be released as 1.0.0-beta.1 (had to drop some Node versions)...

**Comment by pichfl** [comment]
I have updated one of our projects to 1.0.0-beta.1 and the warnings still show. I can provide you with access to the repository if that would help, but it's not public.

**Comment by jrjohnson** [comment]
Same as @pichfl. We updated https://github.com/ilios/frontend to `ember-cli-template-lint@v1.0.0-beta.1` and `ember-template-lint@^1.0.0-beta.2` and are still seeing these errors when running `ember serve`.

**Comment by rwjblue** [comment]
Hmm. I really do not grok what’s going on. I explicitly added tests in Broccoli 2 land and all pass without error or warning. 

Can someone help me with a reproduction?

**Comment by rwjblue** [comment]
Is ilios/frontend a good place to poke around?

**Comment by jrjohnson** [comment]
An update: just upgraded `ilios/frontend` to `ember-cli-template-lint@1.0.0-beta.2` and I'm still seeing these errors:

```
$ yarn list --pattern template-lint
yarn list v1.9.4
├─ ember-cli-template-lint@1.0.0-beta.2
└─ ember-template-lint@1.0.0-beta.5
✨  Done in 0.92s.

```

```
$ yarn start
yarn run v1.9.4
$ ember serve
<---snip some build stuff --->

[API] Warning: The .read and .rebuild APIs will stop working in the next Broccoli version
[API] Warning: Use broccoli-plugin instead: https://github.com/broccolijs/broccoli-plugin
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter
[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter

<-- snip some new linting errors --->
===== 10 Template Linting Errors

[API] Warning: Plugin uses .read/.rebuild API: TemplateLinter

Build successful (59124ms) – Serving on http://localhost:4200/
```

**Comment by rwjblue** [comment]
Ok, thanks for the reminder. Will dig in here, to see what is going on...

**Comment by rwjblue** [comment]
AHA! I just cloned ilios/frontend and dug into the problem. It turns out that the console warnings that you are still seeing are actually coming from [ember-cli-deprecation-workflow](https://github.com/mixonic/ember-cli-deprecation-workflow/blob/299a002a9f35bc926f8f64ea362df459a7bf03f1/generate-deprecations-tree.js#L27) which ironically calls its broccoli plugin base class `TemplateLinter` 🤦‍♂️ therefore throwing us off the trail...

Confirmed by removing `ember-cli-deprecation-workflow` from the package.json of `ilios/frontend`, and seeing no warnings.

Will close this issue and track over in https://github.com/mixonic/ember-cli-deprecation-workflow/issues/52.
